staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history_test.go

```diff
@@ -71,21 +72,25 @@ func TestViewHistory(t *testing.T) {
 			},
 			Spec: appsv1.StatefulSetSpec{Selector: &metav1.LabelSelector{MatchLabels: podStub.ObjectMeta.Labels}, Replicas: &replicas, Template: podStub},
 		}
-
-		ssStub1 = &appsv1.ControllerRevision{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:            "moons",
-				Namespace:       "default",
-				Labels:          map[string]string{"foo": "bar"},
-				OwnerReferences: []metav1.OwnerReference{{"apps/v1", "StatefulSet", "moons", "1993", &trueVar, nil}},
-			},
-			TypeMeta: metav1.TypeMeta{Kind: "StatefulSet", APIVersion: "apps/v1"},
-			Revision: 1,
-		}
 	)
+	stsRawData, err := json.Marshal(ssStub)
```

Comment:
How about tests to verify DaemonSet history with and without revision specified?

Reply:
Added two tests for daemonSet @prachirp (1 for non-zero revision and 1 for zero revision).